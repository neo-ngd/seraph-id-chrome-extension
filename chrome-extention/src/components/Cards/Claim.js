import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import clsx from 'clsx';
import Avatar from '@material-ui/core/Avatar';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardContent from '@material-ui/core/CardContent';
import CardActions from '@material-ui/core/CardActions';
import Collapse from '@material-ui/core/Collapse';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import IconButton from '@material-ui/core/IconButton';
import ReactJson from 'react-json-view';

const useStyles = makeStyles((theme) => ({
  card: {
    backgroundColor: 'transparent',
    boxShadow: 'none',
    borderBottom: 'solid 1px rgba(255, 255, 255, 0.1)'
  },

  expand: {
    transform: 'rotate(0deg)',
    marginLeft: 'auto',
    transition: theme.transitions.create('transform', {
      duration: theme.transitions.duration.shortest,
    }),
  },
  expandOpen: {
    transform: 'rotate(180deg)',
  },
  avatar: {},
}));

export default function RecipeReviewCard({ id, schema, content }) {
  const classes = useStyles();
  const [expanded, setExpanded] = React.useState(false);

  function handleExpandClick() {
    setExpanded(!expanded);
  }


  return (
    <Card className={classes.card}>
      <CardHeader
        action={
          <IconButton
            className={clsx(classes.expand, {
              [classes.expandOpen]: expanded,
            })}
            onClick={handleExpandClick}
            aria-expanded={expanded}
            aria-label="show more"
          >
            <ExpandMoreIcon color={'#FFFFFF'} fontSize="large" />
          </IconButton>
        }
        title={schema}
        subheader={id}
        subheaderTypographyProps={{
          color: '#00BF0B',
          fontSize: "10px"
        }}
        titleTypographyProps={{
          fontSize: "14px"
        }}
      />

      <CardActions disableSpacing></CardActions>
      <Collapse in={expanded} timeout="auto" unmountOnExit>
        <CardContent>
          <ReactJson
            displayObjectSize={false}
            displayDataTypes={false}
            src={content}
          />
        </CardContent>
      </Collapse>
    </Card>
  );
}
