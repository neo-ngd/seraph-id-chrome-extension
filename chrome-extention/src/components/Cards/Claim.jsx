import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import clsx from 'clsx';
import { Card, CardHeader, CardContent, Collapse, IconButton, Box } from '@material-ui/core';
import { Close, ExpandMore } from '@material-ui/icons';
import ReactJson from 'react-json-view';

const useStyles = makeStyles((theme) => ({
  card: {
    backgroundColor: 'transparent',
    boxShadow: 'none',
    borderBottom: 'solid 1px rgba(255, 255, 255, 0.1)'
  },

  expand: {
    transform: 'rotate(0deg)',
    marginLeft: 'auto',
    transition: theme.transitions.create('transform', {
      duration: theme.transitions.duration.shortest,
    }),
    color: theme.palette.text.primary,
    fontSize: '28px',
    padding: 0,
    marginRight: theme.spacing(1),
  },
  expandOpen: {
    transform: 'rotate(180deg)',
  },
  avatar: {},
  title: {
    fontSize: '14px',
  },
  subheader: {
    fontSize: '8px',
    paddingTop: theme.spacing(0.5),
    color: theme.palette.text.hint,
  },
  removeIcon: {
    color: theme.palette.text.secondary,
    padding: 0,
    fontSize: '12px',
  },
  root: {
    padding: '16px 0 8px',
    marginRight: theme.spacing(1),
  },
  action: {
    marginTop: theme.spacing(1) + theme.spacing(0.5),
    paddingRight: theme.spacing(1),
  }
}));

export default function RecipeReviewCard({ id, schema, content }) {
  const classes = useStyles();
  const [expanded, setExpanded] = React.useState(false);

  const handleRemoveClick = () => {
  }

  const renderActionIcons = () => (
    <Box>
      <IconButton
        className={clsx(classes.expand, {
          [classes.expandOpen]: expanded,
        })}
        disableRipple
        onClick={() => setExpanded(!expanded)}
        aria-expanded={expanded}
        aria-label="show more"
      >
        <ExpandMore color="inherit" fontSize="inherit" />
      </IconButton>

      <IconButton
        className={classes.removeIcon}
        disableRipple
        onClick={handleRemoveClick}
        aria-expanded={expanded}
        aria-label="show more"
      >
        <Close color="inherit" fontSize="inherit" />
      </IconButton>
    </Box>
  )


  return (
    <Card className={classes.card}>
      <CardHeader
        action={renderActionIcons()}
        classes={{
          root: classes.root,
          title: classes.title,
          subheader: classes.subheader,
          action: classes.action,
        }}
        title={schema}
        subheader={id}
      />

      <Collapse in={expanded} timeout="auto" unmountOnExit>
        <CardContent>
          <ReactJson
            displayObjectSize={false}
            displayDataTypes={false}
            enableClipboard={false}
            src={content}
          />
        </CardContent>
      </Collapse>
    </Card>
  );
}
