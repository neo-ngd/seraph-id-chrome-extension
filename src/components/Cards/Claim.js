import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import clsx from 'clsx';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardContent from '@material-ui/core/CardContent';
import CardActions from '@material-ui/core/CardActions';
import Collapse from '@material-ui/core/Collapse';
import Avatar from '@material-ui/core/Avatar';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import { red } from '@material-ui/core/colors';
import ShareIcon from '@material-ui/icons/Share';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import MoreVertIcon from '@material-ui/icons/MoreVert';

const useStyles = makeStyles((theme) => ({
  card: {
    maxWidth: 345,
  },

  expand: {
    transform: 'rotate(0deg)',
    marginLeft: 'auto',
    transition: theme.transitions.create('transform', {
      duration: theme.transitions.duration.shortest,
    }),
  },
  expandOpen: {
    transform: 'rotate(180deg)',
  },
  avatar: {
    backgroundColor: red[500],
  },
}));

export default function RecipeReviewCard({ id, schema }) {
  const classes = useStyles();
  const [expanded, setExpanded] = React.useState(false);

  function handleExpandClick() {
    setExpanded(!expanded);
  }

  return (
    <Card className={classes.card}>
      <CardHeader
        avatar={
          <Avatar aria-label="recipe" className={classes.avatar}>
            ID
          </Avatar>
        }
        action={
          <IconButton aria-label="settings">
            <MoreVertIcon />
          </IconButton>
        }
        title={schema}
        subheader={"Unique Identity Number: " + id}
      />

      <CardActions disableSpacing>
        <IconButton aria-label="share">
          <ShareIcon />
        </IconButton>
        <IconButton
          className={clsx(classes.expand, {
            [classes.expandOpen]: expanded,
          })}
          onClick={handleExpandClick}
          aria-expanded={expanded}
          aria-label="show more"
        >
          <ExpandMoreIcon />
        </IconButton>
      </CardActions>
      <Collapse in={expanded} timeout="auto" unmountOnExit>
        <CardContent>
          <Typography paragraph />
          <Typography paragraph />
          <Typography paragraph />
          <Typography />
        </CardContent>
      </Collapse>
    </Card>
  );
}
