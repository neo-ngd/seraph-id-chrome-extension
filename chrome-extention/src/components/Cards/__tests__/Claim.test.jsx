import React from 'react';
import { shallow, mount } from 'enzyme';
import Claim from "../Claim";

const props = {
    id: 11,
    schema: 'test',
    content: {test: 'test'},
    onRemoveClaim: jest.fn(),
};

describe('<Claim />', () => {
    const setState = jest.fn();
    const useStateSpy = jest.spyOn(React, 'useState');
    useStateSpy.mockImplementation((init) => [init, setState]);

    afterEach(() => {
        jest.clearAllMocks();
    });

    it('should render component', () => {
        const wrapper = shallow(<Claim {...props} />);
        expect(wrapper).toMatchSnapshot()
    });

    it('should expand component', () => {
        const wrapper = mount(<Claim {...props} />);
        wrapper.find({'data-test-id': 'claim-expand-button'}).first().simulate('click');
        expect(setState).toHaveBeenCalledWith(true);
    });

    it('should call onRemoveClaim method', () => {
        const wrapper = mount(<Claim {...props} />);
        wrapper.find({'data-test-id': 'claim-remove-button'}).last().simulate('click');
        expect(props.onRemoveClaim).toHaveBeenCalledWith(props.id);
    });
});
