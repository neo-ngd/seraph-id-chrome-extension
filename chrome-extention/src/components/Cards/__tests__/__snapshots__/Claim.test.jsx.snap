// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Claim /> should render component 1`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-card-1"
>
  <WithStyles(ForwardRef(CardHeader))
    action={
      <ForwardRef(StyledComponent)>
        <ForwardRef(WithStyles)
          aria-expanded={false}
          aria-label="show more"
          className="makeStyles-expand-3"
          data-test-id="claim-expand-button"
          disableRipple={true}
          onClick={[Function]}
        >
          <Memo(ExpandMoreIcon)
            color="inherit"
            fontSize="inherit"
          />
        </ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          aria-expanded={false}
          aria-label="show more"
          className="makeStyles-removeIcon-8"
          data-test-id="claim-remove-button"
          disableRipple={true}
          onClick={[Function]}
        >
          <Memo(CloseIcon)
            color="inherit"
            fontSize="inherit"
          />
        </ForwardRef(WithStyles)>
      </ForwardRef(StyledComponent)>
    }
    classes={
      Object {
        "action": "makeStyles-action-10",
        "root": "makeStyles-root-9",
        "subheader": "makeStyles-subheader-7",
        "title": "makeStyles-title-6",
      }
    }
    subheader={11}
    title="test"
  />
  <WithStyles(ForwardRef(Collapse))
    in={false}
    timeout="auto"
    unmountOnExit={true}
  >
    <WithStyles(ForwardRef(CardContent))
      className="makeStyles-cardContent-2"
    >
      <t
        collapseStringsAfterLength={false}
        collapsed={false}
        defaultValue={null}
        displayDataTypes={false}
        displayObjectSize={false}
        enableClipboard={false}
        groupArraysAfterLength={100}
        iconStyle="triangle"
        indentWidth={4}
        name="root"
        onAdd={false}
        onDelete={false}
        onEdit={false}
        onSelect={false}
        shouldCollapse={false}
        sortKeys={false}
        src={
          Object {
            "test": "test",
          }
        }
        style={Object {}}
        theme="rjv-default"
        validationMessage="Validation Error"
      />
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(Collapse))>
</WithStyles(ForwardRef(Card))>
`;
